snforge_std/src/trace.cairo

```diff
@@ -34,3 +34,158 @@ fn get_call_trace() -> CallTrace {
     let mut output = cheatcode::<'get_call_trace'>(array![].span());
     Serde::deserialize(ref output).unwrap()
 }
+
+use core::fmt::{Display, Formatter, Error, Debug};
+
+impl DisplayEntryPointType of Display<EntryPointType> {
+    fn fmt(self: @EntryPointType, ref f: Formatter) -> Result<(), Error> {
+        let str: ByteArray = match self {
+            EntryPointType::Constructor => "Constructor",
+            EntryPointType::External => "External",
+            EntryPointType::L1Handler => "L1 Handler",
+        };
+        f.buffer.append(@str);
+        Result::Ok(())
+    }
+}
+
+impl DisplayCallType of Display<CallType> {
+    fn fmt(self: @CallType, ref f: Formatter) -> Result<(), Error> {
+        let str: ByteArray = match self {
+            CallType::Call => "Call",
+            CallType::Delegate => "Delegate",
+        };
+        f.buffer.append(@str);
+        Result::Ok(())
+    }
+}
+
+impl DisplayCallTrace of Display<CallTrace> {
+    fn fmt(self: @CallTrace, ref f: Formatter) -> Result<(), Error> {
+        Display::fmt(@IndentedCallTrace { struct_ref: self, base_indents: 0 }, ref f).unwrap();
+        Result::Ok(())
+    }
+}
+
+#[derive(Drop)]
+struct IndentedCallTrace {
+    struct_ref: @CallTrace,
+    base_indents: u8,
+}
+
+#[derive(Drop)]
+struct IndentedCallTraceArray {
+    struct_ref: @Array<CallTrace>,
+    base_indents: u8,
+}
+
+#[derive(Drop)]
+struct IndentedEntryPoint {
+    struct_ref: @CallEntryPoint,
+    base_indents: u8,
+}
```

Comment:
Is it possible for it to be some `Indented<T>` in cairo?

Reply:
That's a good point m8